src/components/birth-date-picker/birth-date-picker.tsx

```diff
@@ -0,0 +1,115 @@
+// TODO: use MobX.
+// TODO: filter input?
+// TODO: should we update the state when we receive a different value?
+// TODO: date validation belongs in a separate library.
+
+import React = require("react");
+
+export interface BirthDatePickerProps {
+    value?: Date;
+    minDate?: Date;
+    maxDate?: Date;
+    onChange?: (newValue?: Date) => void;
```

Comment:
Do we want to trigger `onChange` when the input goes from valid to invalid state? If yes, then we can't always pass a date object.